StylableSwiftUI/Classes/StylistIdentifier.swift

```diff
@@ -46,22 +51,40 @@ extension StylistIdentifier: LosslessStringConvertible {
     ///
     /// - parameter description: The string to parse into an identifier
     public init(_ description: String) {
-        let split = description.split(separator: "/").map(String.init)
+        var split = description.split(separator: "/").map(String.init)
 
         let token = split.last ?? ""
+
+        let theme: Theme?
+        if let first = split.first, first.starts(with: Theme.identifierPrefix) {
+            theme = Theme(name: String(first.dropFirst()))
+            split = Array(split.dropFirst())
+        } else {
+            theme = nil
+        }
+
         let path = Path(split.dropLast().joined(separator: "/"))
 
-        self.init(token: token, path: path)
+        self.init(token: token, path: path, theme: theme)
     }
 
     public var description: String {
         let pathDescription = self.path.description
 
         guard !pathDescription.isEmpty else {
+            if let theme = self.theme {
+                return theme.description + "/" + self.token
+            }
             return self.token
         }
 
-        return pathDescription + "/" + self.token
+        let pathAndToken = pathDescription + "/" + self.token
+
+        guard let theme = self.theme else {
+            return pathAndToken
+        }
+
+        return theme.value + "/" + pathAndToken
```

Comment:
Interestingly, you've used `if let` to get the theme on line 75, but used `guard let` to get it on line 83 :)